File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.java
class Solution {
    static class Point {
		double distance;
		int[] point;

		public Point(double distance, int[] point) {
			this.distance = distance;
			this.point = point;
		}
	}
    
    public int[][] kClosest(int[][] points, int k) {
		PriorityQueue<Point> pq = new PriorityQueue<>(Comparator.comparingDouble(a -> a.distance));

		// 좌표 목록으로 거리와 좌표 정보를 Point 객체로 생성하고 우선순위 큐에 넣기
		for (int[] point : points) {
			double dis = Math.sqrt(Math.pow((0 - point[0]), 2) + Math.pow((0 - point[1]), 2));
			pq.add(new Point(dis, point));
		}

		int[][] answer = new int[k][];

        for(int i = 0; i< k; i++){
            answer[i] = pq.poll().point;
        }
        
		return answer;
    }
}

File: 0973-k-closest-points-to-origin/0973-k-closest-points-to-origin.kt
class Solution {
    data class Point(var distance: Double, var point: IntArray)
    
    fun kClosest(points: Array<IntArray>, k: Int): Array<IntArray> {
        val pq = PriorityQueue<Point>(compareBy { a -> a.distance })

        for (point in points) {
            val dis = point[0].toDouble() * point[0] + point[1].toDouble() * point[1]
            pq.add(Point(dis, point))
        }

        val answer: Array<IntArray> = Array(k) { IntArray(2) { 0 } }
        for (i in 0 until k) {
            answer[i] = pq.poll().point
        }

        return answer
    }
}